docs/guide/docs/main/overview.md

```diff
@@ -1,112 +1,154 @@
-3. It **decides** on what to respond back to the user
+## Anatomy of a chatbot
+A chatbot is quite simple:
 
-#### Lifecycle
+1. It receives messages from Messaging **channels**.
+2. It **processes** these messages to understand, translate, or escalate underlying information.
+3. It **decides** how to respond to the user
+![High-Level Diagram](../assets/hld.png)
+The red dots are the developer extension points. These indicate where a chatbot developer can inject code to customize their chatbot.
 
-![High-Level Diagram](assets/hld.png)
+### Channels
+Think of a channel in the context of communication. Using this definition, we can describe a channel as any chat platform which connects your chatbot to the intended user, for example, Facebook Messenger. Botpress provides pre-built modules which allow your chatbot to send and receive messages from the following chat platforms:
+ 
+ - Website Embedded Chat
+ - Facebook Messenger
+ - Twilio
+ - Slack
+ - Telegram
+ - Microsoft Teams
+ - Sunshine Conversations
+ - Vonage
 
-<br/>
+> In Botpress, channels are installed and configured individually and locally, giving you complete control over the data transmitted between your chatbot and the chat platforms. Botpress does not proxy the messages to a 3rd party service.
 
-In red are the developer extension points, i.e. where you will be able to inject code to customize the bot.
+Behind the scenes, Botpress implements a queuing mechanism that processes ingoing and outgoing messages sequentially. If a message fails to be processed or sent for some reason, Botpress will retry the message processing sequence before raising an error to the developer and chatbot administrator.
 
-## Channels
+### Processing
+Messages received from messaging channels are then processed. During processing, Botpress extracts data from the text sent by a chat channel user and processes it in the following ways:
 
-A channel is a module that allows your bot to send and receive messages from a specific chat platform (Facebook Messenger, Telegram, etc.).
+#### NLU
+**Natural Language Understanding** (or **NLU**) allows your chatbot to process messages into structured data which your chatbot can understand. The main tasks of the NLU engine are:
 
-> Note: Unlike most other bot platforms, channels are installed and configured individually and locally, which means you have full control over the data that is transmitted between your bot and the chat platforms. Botpress does not proxy the messages to a 3rd party service as Microsoft Bot Framework does.
+- **Intent Recognition**: recognizing what the user wants.
+- **Entity Extraction**: extracting structured information from messages like dates, time, cities, names, and more.
+- **Slot tagging**: identify necessary **parameters** to fulfill given tasks.
+- **Language Identification**: knowing in which language the user is writing.
+- **Spell Checking**: giving you one less thing to worry about by making sure your chatbot is fluent.
```

Comment:
I'm not sure this is a proper description of what spell checking is ..

Spell checking makes sure users' text input are are spelled correctly. I.E typo fix, etc.

Reply:
Okay, let me make that clearer.